lib/service/storage/elasticsearch.js

```diff
@@ -1819,12 +1819,12 @@ class ElasticSearch extends Service {
       index: await this._getIndice(index, collection)
     };
 
-    const mutex = new Mutex(`hiddenCollection/delete/${index}`);
+    const mutex = new Mutex(`hiddenCollection/${index}`);
     try {
-      await this._client.indices.delete(esRequest);
-
       await mutex.lock();
 
+      await this._client.indices.delete(esRequest);
+
       if (! await this._hasHiddenCollection(index)) {
         await this._createHiddenCollection(index);
```

Comment:
A bit tricky but `deleteCollection` uses the mutex `hiddenCollection/${index}` then later calls `_createHiddenCollection` which also use the mutex `hiddenCollection/${index}` which is already locked at the moment it is called, which causes your test to be stuck in an infinite loop